pkg/controllers/resources/pods/conditions.go

```diff
@@ -0,0 +1,81 @@
+package pods
+
+import (
+	synccontext "github.com/loft-sh/vcluster/pkg/controllers/syncer/context"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+// UpdateConditions adds/updates new/old conditions in the physical Pod
+func UpdateConditions(ctx *synccontext.SyncContext, physicalClusterClient kubernetes.Interface, physicalPod *corev1.Pod, virtualPod *corev1.Pod) error {
+	// check if the readinessGates are added to vPod
+	if len(virtualPod.Spec.ReadinessGates) > 0 {
+		newCustomConditions, existingCustomConditions := getCustomConditions(physicalPod, virtualPod)
+		// check if newConditions need to be added.
+		if len(newCustomConditions) > 0 {
+			physicalPod.Status.Conditions = append(physicalPod.Status.Conditions, newCustomConditions...)
+			_, err := physicalClusterClient.CoreV1().Pods(physicalPod.Namespace).UpdateStatus(ctx.Context, physicalPod, metav1.UpdateOptions{})
+			if err != nil {
+				return err
+			}
+		}
+
+		// check if existingConditions need to be updated.
+		if len(existingCustomConditions) > 0 {
+			for index, condition := range existingCustomConditions {
+				physicalPod.Status.Conditions[index] = condition
+			}
+			_, err := physicalClusterClient.CoreV1().Pods(physicalPod.Namespace).UpdateStatus(ctx.Context, physicalPod, metav1.UpdateOptions{})
```

Comment:
Prefer ctx.PhysicalClient here and condense this into a single status update call only when the conditions have actually changed